app/scripts/directives/notifications/notificationDrawerWrapper.js

```diff
@@ -154,11 +154,12 @@
       var formatAndFilterEvents = function(eventMap) {
         var filtered = {};
         ensureProjectGroupExists(filtered, $routeParams.project);
-        _.each(eventMap, function(event) {
+        _.each(eventMap, function(event, eventId) {
           if(EventsService.isImportantEvent(event) && !EventsService.isCleared(event)) {
             ensureProjectGroupExists(filtered, event.metadata.namespace);
             filtered[event.metadata.namespace].notifications.push({
               unread:  !EventsService.isRead(event),
+              uid: eventId,
```

Comment:
I'd prefer `event.metadata.uid` here so it doesn't matter how the map is keyed. uid seems safest since it's globally unique.

Reply:
Agree, updating this.